sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/models/ShareEnabledProtocol.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.share.models;
+
+import com.azure.storage.common.implementation.Constants;
+
+/**
+ * Represents protocols that can be set on a share.
+ */
+public class ShareEnabledProtocol {
```

Comment:
should this be named `ShareEnabledProtocols` ? As far as I understand one can have multiple of them enabled when this "enum" expands.

I'd rather use `enum` and `EnumSet` to model this. See https://github.com/Azure/azure-sdk-for-java/blob/4da5d307be3b618c7c989e76b13cf1e96e9dbb17/sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/models/NtfsFileAttributes.java for example.

Reply:
I can change the name

I thought about EnumSet, but it seems like this this collection of booleans is more common in the sdk for representing what is a flags enum in .net, so I went with this. Might be a question for ApiView time?

Reply:
Let's see what @alzimmermsft thinks.

Reply:
In the future are there plans to support both SMB and NFS concurrently? Would additional protocols be added?

If only one protocol is expected using an enum value would help guide usage to non-error cases but if support will be expanded to multiple it may result in a poor API growth.

Reply:
Possible values in the future at REST API:
```
"SMB;NFS"
"SMBv2;SMBv3;NFSv4"
```
see original discussion https://github.com/Azure/azure-sdk-for-net/pull/16123#discussion_r508775909

So looks like we stick to current design.